### PR TITLE
Wait for any activeTransactions to finish during Create

### PR DESCRIPTION
**Pull request by hairyhenderson**
This adds an additional wait phase to make sure any active Transactions
are completed before Create tries to SSH in to the new host. This is
sometimes necessary because SSH can become available before SoftLayer
is fully done setting up the host and strange things can happen...

Signed-off-by: Dave Henderson <Dave.Henderson@ca.ibm.com>

### CODE REVIEW

File: drivers/softlayer/driver.go [hunk 1]
```diff
@@ -305,11 +313,41 @@ func (d *Driver) Create() error {
...
 			if s == state.Running {
 				break
+			} else {
+				log.Debugf("Still waiting - state is %s...", s)
+			}
+			time.Sleep(2 * time.Second)
+		}
+	}
+
+	waitForSetupTransactions := func() {
```
Comment (nathanleclaire) [inline drivers/softlayer/driver.go]: Any particular reason this is done for using a closure here?  I'd really prefer to move this and the other closures into real methods / helper functions.

Comment (hairyhenderson) [inline drivers/softlayer/driver.go]: Nope. Was just trying to be consistent with the code above. I thought of refactoring and moving the three out, but wanted to minimize the changes.

Comment (nathanleclaire) [inline drivers/softlayer/driver.go]: If you wanted to do so, I'd be OK with it!

Comment (hairyhenderson) [inline drivers/softlayer/driver.go]: ok, done :)

**Comment by nathanleclaire** [comment]
FYI eventually we will be moving some of that logic out of the driver (shouldn't have to get the IP, wait for SSH, or run `apt-get` commands inside the `Create` method) but until then - LGTM

**Comment by nathanleclaire** [comment]
cc @ehazlett

**Comment by ehazlett** [comment]
+1 to what @nathanleclaire said -- eventually these are planned to be moved out.  

LGTM

**Pull request merged**